### PR TITLE
Update lint setup

### LINKED ISSUE #27: Codestyle tooling conflict
**Issue**
As a developer, I would like to have a single point of truth for code style standards so that I do not have points of conflicts

It's nice that we now have a pre-commit setup for running certain hooks before commit and/or push. But the pre-commit uses ruff while github action uses flake8 and pylint and ignores pre-commit. This creates conflict when a developer decides to use the pre-commit hooks in their local development before pushing. 

**Definition of done**
There is a single point of truth, preferably ruff.

### PR DESCRIPTION
**Pull request by larsevj**
Closes #27 

**Approach**
Adds pylint to pre-commit, and removes flake8 from github style test. Adds ruff rules similar to ert.

### CODE REVIEW

**Comment by sregales-TNO** [comment]
I created issue #27 without realizing you had PR around the same subject in progress.
Will this PR cover this issue. 
If this is the case, It would be better to reduce the code style to ruff , mypy, and maybe in the future typegaurd.

example pyproject.toml
```toml
[tool.ruff]  # https://github.com/charliermarsh/ruff
fix = true
line-length = 100
src = ["src", "tests"]
target-version = "py312"

[tool.ruff.lint]
select = [
    "D",  # pydocstyle
    "E",  # pycodestyle errors
    "W",  # pycodestyle warnings
    "F",  # pyflakes
    "I",  # isort
    "C",  # flake8-comprehensions
    "B",  # flake8-bugbear
]
ignore = [
    "E501",  # line too long, handled by black
    "B008",  # do not perform function calls in argument defaults
    "C901",  # too complex
    "W191",  # indentation contains tabs
    "F405",  # undefined * import
    "E722",  # catch all except
]
unfixable = ["ERA001", "F401", "F841", "T201", "T203"]

[tool.ruff.lint.pycodestyle]
max-doc-length = 100

[tool.ruff.lint.pydocstyle]
convention = "google"

[tool.ruff.lint.per-file-ignores]
# Ignore all files that end in `_test.py`.
"*_test.py" = ["D"]
"tests/**/__init__.py" = ["D"]

[tool.mypy]  # https://mypy.readthedocs.io/en/latest/config_file.html
junit_xml = "reports/mypy.xml"
strict = true
disallow_subclassing_any = false
disallow_untyped_decorators = false
ignore_missing_imports = true
pretty = true
show_column_numbers = true
show_error_codes = true
show_error_context = true
warn_unreachable = true
exclude = ["conftest.py$"]
```
and this will eliminate all the other configuration files.

**Comment by sregales-TNO** [comment]
> Maybe ruff could replace pylint too, but I believe there is some discrepancy between the two for the time being.

what are we trying to capture with pylint that we can't with ruff. is it worth having to points of truth?

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -10,8 +10,16 @@ repos:
     -   id: trailing-whitespace
 
 -   repo: https://github.com/astral-sh/ruff-pre-commit
-    rev: v0.3.5
+    rev: v0.4.8
```
Comment (sregales-TNO) [inline .pre-commit-config.yaml:13]: we can eliminate the need of manually adding the next version with the following lines

```yaml
ci:
  autofix_commit_msg: 🎨 [pre-commit.ci] Auto format from pre-commit.c
  autoupdate_commit_msg: ⬆ [pre-commit.ci] pre-commit au
  ```
  emoji can be omitted if you wish

**Review by DanSava — APPROVED** [review]
Nice one! :1st_place_medal:

**Pull request merged**